internal/k8s/nodeGroup.go

```diff
@@ -0,0 +1,250 @@
+package k8s
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/topfreegames/kaas-management-api/util/clientError"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterapiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	clusterapiexpv1beta1 "sigs.k8s.io/cluster-api/exp/api/v1beta1"
+)
+
+type NodeGroup struct {
+	Name               string
+	Cluster            string
+	Environment        string
+	Region             string
+	InfrastructureName string
+	InfrastructureKind string
+	Replicas           *int32
+}
+
+// GetMachinePool Returns a Machinepool CR from a specific cluster
+func (k Kubernetes) GetMachinePool(clusterName string, nodeGroupName string) (*clusterapiexpv1beta1.MachinePool, error) {
+
+	client := k.K8sAuth.DynamicClient
+
+	resource := client.Resource(MachinePoolSchemaV1beta1)
+	machinePoolRaw, err := resource.Namespace(clusterName).Get(context.TODO(), nodeGroupName, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil, clientError.NewClientError(err, clientError.ResourceNotFound, fmt.Sprintf("The requested machinepool %s was not found for the cluster %s!", nodeGroupName, clusterName))
+		} else if statusError, isStatus := err.(*errors.StatusError); isStatus {
+			return nil, fmt.Errorf("Error getting machinepool: %v\n", statusError.ErrStatus.Message)
+		}
+		return nil, fmt.Errorf("Internal server clientError: %v\n", err)
+	}
+
+	var machinePool clusterapiexpv1beta1.MachinePool
+	machinePoolRawJson, err := machinePoolRaw.MarshalJSON()
+	if err != nil {
+		return nil, fmt.Errorf("could not Marshal machinepool response: %v", err)
+	}
+
+	err = json.Unmarshal(machinePoolRawJson, &machinePool)
+	if err != nil {
+		return nil, fmt.Errorf("could not Unmarshal machinepool JSON into clusterAPI list: %v", err)
+	}
+
+	//err := sanitizeMachinePool(&machinePool)
```

Comment:
Why is this block of code commented?

Reply:
It should be removed.   
CAPI sanitizes the machinepool/machinedeployment, we should not need to confirm what's already confirmed.